pkg/odo/cli/debug/info.go

```diff
@@ -46,7 +45,7 @@ func NewInfoOptions() *InfoOptions {
 
 // Complete completes all the required options for port-forward cmd.
 func (o *InfoOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
```

Comment:
Why was the other check removed? This will trigger `NewDevfileContext` even for s2i components.
```
[mrinaldas@localhost project]$ odo create nodejs --s2i
Experimental mode is enabled, use at your own risk

Validation
 ✓  Validating component [2s]

Please use `odo push` command to create the component with source deployed
[mrinaldas@localhost project]$ odo debug info
 ✗  The current directory does not represent an odo component. Use 'odo create' to create component here or switch to directory with a component
```

Reply:
I didnt realize we had a `s2i` flag for experimental, I was wondering what the point of the check was. Thanks for showing it out, will update it!